### ISSUE TITLE
Possible inconcistency in handling DAE equation order

### ISSUE THREAD

**Issue by lamorton** [SciML/ModelingToolkit.jl#1195]
There's a bunch of logic in this constructor [`ODESystem(eqs)`](https://github.com/SciML/ModelingToolkit.jl/blob/dd647204448a33ab27c7cc2980ce641148506b08/src/systems/diffeqs/odesystem.jl#L164-L181) to re-order the equations such that the differential ones come first. For some reason, this is not done in the other constructors (ex: [`ODESystem(eqs,iv,vars,ps)`](https://github.com/SciML/ModelingToolkit.jl/blob/dd647204448a33ab27c7cc2980ce641148506b08/src/systems/diffeqs/odesystem.jl#L99)). That seems like a potential mistake, because there's no reason one couldn't use those others to construct a DAE as well, but then the equations would not be re-ordered. 

I attempted moving the logic to the inner constructor, but the current implementation of the 'sort' drops any variables that do not appear in an equation, which causes a problem for [this test](https://github.com/SciML/ModelingToolkit.jl/blob/dd647204448a33ab27c7cc2980ce641148506b08/test/odesystem.jl#L352-L359).

Any idea what the right thing to do is? I'd like to modify that test so it doesn't rely on a pathological system with a variable that has no governing equation, but are people using that somewhere? I feel like that case should really be handled by making the variable a parameter instead.

**Comment by ChrisRackauckas** [comment]
It should all be done in an inner constructor. It's kind of a mess right now, and https://github.com/SciML/ModelingToolkit.jl/pull/1164 was one attempt, but @YingboMa has another idea.

**Comment by lamorton** [comment]
I see, that reordering will fail anyway if the differential isn't the LHS. OK, I'm not going to wait on this with #1156.

**Comment by YingboMa** [comment]
The multi-arg constructor is called very often, so it's deliberate that we don't have equation sorting in there.

**Comment by lamorton** [comment]
I see. So should we just copy the logic to the other relevant constructors? In that case, I would put the units checking before the reorder in the outer constructors, and either remove it from the inner constructor or put a flag to bypass it in the event it’s already done by the outer constructors.

**Comment by lamorton** [comment]
Stale.
